metadata-ingestion/src/datahub/cli/docker_check.py

```diff
@@ -1,122 +1,203 @@
+import enum
 import os
 from contextlib import contextmanager
-from typing import Iterator, List, Optional, Tuple
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Optional
 
 import docker
+import docker.errors
+import docker.models.containers
+
+from datahub.configuration.common import ExceptionWithProps
 
 REQUIRED_CONTAINERS = [
-    "elasticsearch-setup",
     "elasticsearch",
     "datahub-gms",
     "datahub-frontend-react",
-    "datahub-upgrade",
-    "schema-registry",
     "broker",
-    "zookeeper",
-    # These two containers are not necessary - only helpful in debugging.
-    # "kafka-topics-ui",
-    # "schema-registry-ui",
-    # "kibana",
-    # "kafka-rest-proxy",
-    # "datahub-mce-consumer",
-    # "datahub-mae-consumer"
 ]
 
+# We expect these containers to exit 0, while all other containers
+# are expected to be running and healthy.
 ENSURE_EXIT_SUCCESS = [
     "kafka-setup",
     "elasticsearch-setup",
     "mysql-setup",
     "datahub-upgrade",
 ]
 
+# If present, we check that the container is running / exited properly.
 CONTAINERS_TO_CHECK_IF_PRESENT = [
-    # We only add this container in some cases, but if it's present, we
-    # definitely want to check that it exits properly.
     "mysql",
     "mysql-setup",
     "cassandra",
     "cassandra-setup",
     "neo4j",
+    "elasticsearch-setup",
```

Comment:
Question -- Why does this need to appear in both ENSURE_EXIT_SUCCESS and CONTAINERS_TO_CHECK_IF_PRESENT?

Reply:
Also does kafka-setup belong in here as well? (If needed in both places)

Reply:
yes adding kafka-setup